### PR TITLE
modify(common): Refactor help-keyman-com.sh script for uploading help files

### LINKED ISSUE #4424: change(common): Refactor help-keyman-com.sh for uploading other platform help
Now that more of the markdown help for Keyman products will be configured in this repo
#4397 Android
#4414 Linux

we should refactor `windows/src/buildtools/help-keyman-com.sh` to upload help these other platforms too.

* Move to resources/build/help-keyman-com.sh
* Use `$PLATFORM` environment variable to determine help and destination paths
* Use `$TIER` to determine base branch (`staging` for alpha and beta tiers)
*

### PR DESCRIPTION
**Pull request by darcywong00**
Fixes #4424 
This refactors help-keyman-com.sh for uploading help files to help.keyman.com across the Keyman platforms

## Preliminary cleanup for /android/help/
* move build-help.sh and htmlink.lua out of `android/help/` (simplifies things so the entire help/ folder gets copied)
* Remove context/menu_*.md files (they were already replaced with context/menu-*.md)

Commit 56b4b4c23b1f0d10cdb438e073906267a6d1b204 has the refactoring of help-keyman-com.sh

Some "artistic liberties" in the refactor:
* I chose to use an environment var `PLATFORM` to determine which Keyman help gets uploaded. Since the platform determines the upload paths, we can't just target `all`. (Maybe this should be a command-line parameter?)
* I elected to use a base branch `staging` instead of `master` for help.keyman.com for pre-release tiers.
* I went ahead and coded the paths for all the platforms, even though not all the help is checked into this repo yet.

### CODE REVIEW

File: resources/build/help-keyman-com.sh [hunk 1]
```diff
@@ -0,0 +1,163 @@
...
+#
+# In this script we update the git repo and copy the Keyman for [Platform]
+# documentation over. This should be run after a full build.
+#
+
+#
+# These are passed via environment:
+#
+# HELP_KEYMAN_COM = the home of the help.keyman.com repository
+# PLATFORM = platform of the Keyman help to upload
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:31]: While `HELP_KEYMAN_COM` can be environment, I think `PLATFORM` should be a parameter to the script.

File: resources/build/help-keyman-com.sh [hunk 2]
```diff
@@ -0,0 +1,163 @@
...
+    linux)
+      helppath=$KEYMAN_ROOT/linux/help
+      dstpath="$HELP_KEYMAN_COM/products/linux/$VERSION_RELEASE"
+      ;;
+    mac)
+      helppath=$KEYMAN_ROOT/mac/help
+      dstpath="$HELP_KEYMAN_COM/products/mac/$VERSION_RELEASE"
+      ;;
+    windows)
+      helppath=$KEYMAN_ROOT/windows/bin/help/md/desktop
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:88]: 🤣 love how the Windows help is so convoluted in comparison. Might be worth adding a comment here to note that the Windows build must have completed before this script can be run.

**Review by mcdurdin — APPROVED** [review]
LGTM. Nice work. Can you just change over `PLATFORM` to be a parameter and include the usual script help if no parameters (or `--help` etc) passed in?

File: resources/build/help-keyman-com.sh [hunk 3]
```diff
@@ -0,0 +1,202 @@
...
+
+if [ -t 2 ]; then
+  t_red=$'\e[1;31m'
+  t_grn=$'\e[1;32m'
+  t_yel=$'\e[1;33m'
+  t_blu=$'\e[1;34m'
+  t_mag=$'\e[1;35m'
+  t_cyn=$'\e[1;36m'
+  t_end=$'\e[0m'
+fi
```
Comment (darcywong00) [inline resources/build/help-keyman-com.sh:68]: had to add this block for the warning in l.114

I guess we don't have a standard util.sh for terminal colors

File: resources/build/help-keyman-com.sh [hunk 4]
```diff
@@ -0,0 +1,202 @@
...
+# Main
+#
+
+platform=
+
+# Process command-line arguments
+while [[ $# -gt 0 ]] ; do
+  key="$1"
+  case $key in
+    -help|-h)
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:181]: Can we use `--help` for long parameter name?

File: resources/build/help-keyman-com.sh [hunk 5]
```diff
@@ -0,0 +1,202 @@
...
+      display_usage
+      exit
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:183]:
```suggestion
      exit 0
```

It's not an error, I guess...

File: resources/build/help-keyman-com.sh [hunk 6]
```diff
@@ -0,0 +1,202 @@
...
+      ;;
+    android | ios | linux | linux | mac | windows)
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:185]:
```suggestion
    android | ios | linux | mac | windows)
```

File: resources/build/help-keyman-com.sh [hunk 7]
```diff
@@ -0,0 +1,202 @@
...
+  exit 1
+fi
+
+function display_usage {
+  echo "Usage: $0 [platform]"
+  echo "       $0 -help"
+  echo
+  echo "  platform should be one of: android, ios, linux, mac, windows."
+  echo "  -help               displays this screen and exits"
+  exit 1
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:53]: Probably drop this `exit` here as the cases in the main program handle it.

File: resources/build/help-keyman-com.sh [hunk 8]
```diff
@@ -0,0 +1,202 @@
...
+    -help|-h)
+      display_usage
+      exit
+      ;;
+    android | ios | linux | linux | mac | windows)
+      platform=$key
+      ;;
+    *)
+      echo "$0: invalid option: $key"
+      display_usage
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:190]:
```suggestion
      echo "$0: invalid option: $key"
      display_usage
      exit 1
```

File: resources/build/help-keyman-com.sh [hunk 9]
```diff
@@ -0,0 +1,202 @@
...
+  esac
+  shift # past the processed argument
+done
+
+if [ -z ${platform} ]; then
+  display_usage
+fi
```
Comment (mcdurdin) [inline resources/build/help-keyman-com.sh:197]:
```suggestion
  display_usage
  exit 1
fi
```

**Review by mcdurdin — APPROVED** [review]
Minor changes, but lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 14.0.240-beta](https://keyman.com/downloads/releases/beta/14.0.240)

**Comment by mcdurdin** [comment]
Looks like this has failed: https://build.palaso.org/viewLog.html?buildId=252214&buildTypeId=Keyman_Build&tab=buildLog&_focus=15177

```
[05:30:13]Step 8/9: Upload new Keyman Desktop help to help.keyman.com (Command Line) (1s)
[05:30:13][Step 8/9] Starting: C:\BuildAgent\temp\agentTmp\custom_script1399629416138801554.cmd
[05:30:13][Step 8/9] in directory: C:\BuildAgent\work\82cac59564031c74\keyman\resources\build
[05:30:14][Step 8/9] Uploading Keyman for windows documentation to help.keyman.com
[05:30:14][Step 8/9] Committing and pushing updated Keyman for windows documentation
[05:30:14][Step 8/9] /c/BuildAgent/work/82cac59564031c74/help.keyman.com /c/BuildAgent/work/82cac59564031c74/keyman/resources/build
[05:30:14][Step 8/9] fatal: 'staging' is not a commit and a branch 'auto/windows-help-14.0.240-beta' cannot be created from it
[05:30:14][Step 8/9] fatal: ../../../help.keyman.com/products/windows/14.0: '../../../help.keyman.com/products/windows/14.0' is outside repository
[05:30:14][Step 8/9] Process exited with code 1
[05:30:14][Step 8/9] Process exited with code 1 (Step: Upload new Keyman Desktop help to help.keyman.com (Command Line))
[05:30:14][Step 8/9] Step Upload new Keyman Desktop help to help.keyman.com (Command Line) failed
```

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 15.0.16-alpha](https://keyman.com/downloads/releases/alpha/15.0.16)

**Pull request merged**